examples/irsa_autoscale_refresh/main.tf

```
@@ -0,0 +1,165 @@
+provider "aws" {
+  region = local.region
+}
+
+locals {
+  name            = "ex-${replace(basename(path.cwd), "_", "-")}"
+  cluster_version = "1.21"
+  region          = "eu-west-1"
+
+  tags = {
+    Example    = local.name
+    GithubRepo = "terraform-aws-eks"
+    GithubOrg  = "terraform-aws-modules"
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.eks.cluster_id
+}
+
+################################################################################
+# EKS Module
+################################################################################
+
+module "eks" {
+  source = "../.."
+
+  cluster_name                    = local.name
+  cluster_version                 = local.cluster_version
+  cluster_endpoint_private_access = true
+  cluster_endpoint_public_access  = true
+
+  vpc_id     = module.vpc.vpc_id
+  subnet_ids = module.vpc.private_subnets
+
+  enable_irsa = true
+
+  # Self Managed Node Group(s)
+  self_managed_node_groups = {
+    refresh = {
+      max_size     = 5
+      desired_size = 1
+
+      instance_types                  = ["m5.large", "m5n.large", "m5zn.large", "m6i.large", ]
+      create_launch_template          = true
+      launch_template_name            = "refresh"
+      launch_template_default_version = true
+
+      instance_refresh = {
+        strategy = "Rolling"
+        preferences = {
+          checkpoint_delay       = 600
+          checkpoint_percentages = [35, 70, 100]
+          instance_warmup        = 300
+          min_healthy_percentage = 50
+        }
+        triggers = ["tag"]
+      }
+
+      propogate_tags = [{
+        key                 = "aws-node-termination-handler/managed"
+        value               = ""
+        propagate_at_launch = true
+      }]
+    }
+  }
+
+  tags = merge(local.tags, { Foo = "bar" })
+}
+
+################################################################################
+# aws-auth configmap
+# Only EKS managed node groups automatically add roles to aws-auth configmap
+# so we need to ensure fargate profiles and self-managed node roles are added
+################################################################################
+
+data "aws_eks_cluster_auth" "this" {
+  name = module.eks.cluster_id
+}
+
+locals {
+  kubeconfig = yamlencode({
+    apiVersion      = "v1"
+    kind            = "Config"
+    current-context = "terraform"
+    clusters = [{
+      name = "${module.eks.cluster_id}"
+      cluster = {
+        certificate-authority-data = "${module.eks.cluster_certificate_authority_data}"
+        server                     = "${module.eks.cluster_endpoint}"
+      }
+    }]
+    contexts = [{
+      name = "terraform"
+      context = {
+        cluster = "${module.eks.cluster_id}"
+        user    = "terraform"
+      }
+    }]
+    users = [{
+      name = "terraform"
+      user = {
+        token = "${data.aws_eks_cluster_auth.this.token}"
```

```suggestion
        token = data.aws_eks_cluster_auth.this.token
```

Similar fix in other places above.

---

ah yes, fixed in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1680/commits/e9d94763b53ccb21f9bc965cc8cdafd42111e8f3